scipy/signal/tests/test_short_time_fft.py

```diff
@@ -541,7 +541,9 @@ def test_border_values_exotic():
     assert SFT.lower_border_end == (0, 0)
 
     SFT = ShortTimeFFT(np.flip(w), hop=20, fs=1)
-    assert SFT.upper_border_begin(4) == (0, 0)
+    assert SFT.upper_border_begin(4) == (16, 1)
+    assert SFT.upper_border_begin(3) == (16, 1)
+    assert SFT.upper_border_begin(2) == (16, 1)
```

Comment:
```suggestion
    assert SFT.upper_border_begin(5) == (16, 1)
    assert SFT.upper_border_begin(23) == (36, 2)
    assert SFT.upper_border_begin(24) == (36, 2)
    assert SFT.upper_border_begin(25) == (36, 2)
```
Due to PR #22792 `n` needs to be ≥ `len(w)/2`.